feat(vscode-extensions): Add section about how to use Eclipse Che API (#1867)
codenvy/che-docs: modules/contributor-guide/partials/ref_che-theia-plug-in-api.adoc

| workspace | Allows to work with the {prod-short} workspace API; provides functionality to access workspaces, their lifecycle, and user settings.
|===

> [id="using-the-{prod-id-short}--in-plug-ins_{context}"]

Reviewer: ```suggestion
[id="using-the-{prod-id-short}-in-plug-ins_{context}"]
```

Author: thanks, applied

== Using the `che` {orch-namespace} in plug-ins

VS Code extensions::
+

> . Adding the `@eclipse-che/plugin` dependency for backend plug-ins to `package.json` file is a possibility.

Reviewer: ```suggestion
. Adding the `@eclipse-che/plugin` dependency for backend plug-ins to a `package.json` file is a possibility.
```

Author: thanks, applied

> . But for quick calls, it's simpler to use VS Code extension mechanism because it does not require to rely on 3rd party dependency or try and catching the error.

Reviewer: ```suggestion
. For quick calls, it is simpler to use VS Code extension mechanism because it does not require relying on a third-party party dependency or trying and catching the error.
```

Author: thanks, applied
